src/graphnet/data/i3extractor.py

```diff
@@ -289,12 +289,22 @@ def __call__(self, frame, padding_value=-1) -> dict:
         }
 
         if frame['I3EventHeader'].sub_event_stream == 'InIceSplit': #only inicesplit p frames have filters calculated
-            output.update({
-                'DeepCoreFilter_13': int(bool(frame['FilterMask']['DeepCoreFilter_13'])),
-                'CascadeFilter_13': int(bool(frame['FilterMask']['CascadeFilter_13'])),
-                'MuonFilter_13': int(bool(frame['FilterMask']['MuonFilter_13'])),
-                'OnlineL2Filter_17': int(bool(frame['FilterMask']['OnlineL2Filter_17'])),
-            })
+
+            if frame_has_key(frame,key='FilterMask'):
+                output['DeepCoreFilter_13'] = int(bool(try_get_key(frame["FilterMask"],'DeepCoreFilter_13',default_value=-1)))
+                output['CascadeFilter_13'] = int(bool(try_get_key(frame["FilterMask"],'CascadeFilter_13',default_value=-1)))
+                output['MuonFilter_13'] = int(bool(try_get_key(frame["FilterMask"],'MuonFilter_13',default_value=-1)))
+                output['OnlineL2Filter_17'] = int(bool(try_get_key(frame["FilterMask"],'OnlineL2Filter_17',default_value=-1)))
```

Comment:
```suggestion
                output.update({
                    'DeepCoreFilter_13': int(bool(try_get_key(frame["FilterMask"], 'DeepCoreFilter_13', default_value=-1))),
                    'CascadeFilter_13': int(bool(try_get_key(frame["FilterMask"], 'CascadeFilter_13', default_value=-1))),
                    'MuonFilter_13': int(bool(try_get_key(frame["FilterMask"], 'MuonFilter_13', default_value=-1))),
                    'OnlineL2Filter_17': int(bool(try_get_key(frame["FilterMask"], 'OnlineL2Filter_17', default_value=-1))),
                })
```

Reply:
@BozianuLeon, I am a bit puzzled by these four statements. The syntax is: `int(bool(try_get_key(frame["FilterMask"], key, default_value=-1)))`. What is the expected type of the entries in `frame["FilterMask"]`? Do we need the cast to `bool` if we want to return `int`, especially if the default value is `-1`?